### PR TITLE
Feat: Added email template for various actions in team section 

### PR DESCRIPTION
**Pull request by Abhishek9503**
## Description

 
#1199  #1202 #1203 #1207 #1211 #1208  #1209 #1210 #1213 #1218 #1217 
In this PR i have added the email template for  team updates like inviting a member , removing a member , leaving a team ,  Changing designation in team , workspace invitation  etc.

## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [x] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build

## Have you tested locally?

- [x] 👍 yes
- [ ] 🙅 no, because I am lazy

## Added tests?

- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md

### CODE REVIEW

File: src/modules/identity/services/team-user.service.ts [hunk 1]
```diff
@@ -461,6 +480,18 @@ export class TeamUserService {
     return response;
   }
 
+  async OwnerDetails(
```
Comment (Astitva877) [inline src/modules/identity/services/team-user.service.ts:483]: change name to ownerDetails or rename it something more understandable like getOwnerDetails

File: src/modules/identity/services/team-user.service.ts [hunk 2]
```diff
@@ -517,6 +548,105 @@ export class TeamUserService {
...
+        teamName: teamName,
+        sparrowEmail: this.configService.get("support.sparrowEmail"),
+      },
+      subject: `Team Member Update: ${MemberName} has left ${teamName}`,
+    };
+    const promise = [transporter.sendMail(mailOptions)];
+    await Promise.all(promise);
+  }
+
+  async removeUserEmail(
```
Comment (Astitva877) [inline src/modules/identity/services/team-user.service.ts:610]: add jsdoc

**Review by Astitva877 — CHANGES_REQUESTED** [review]
add await wherever promise is returning and add jsdoc also

**Comment by Abhishek9503** [comment]
> add await wherever promise is returning and add jsdoc also
Resovled

File: src/modules/workspace/services/workspace.service.ts [hunk 3]
```diff
@@ -759,6 +781,22 @@ export class WorkspaceService {
       new ObjectId(payload.workspaceId),
       updatedWorkspaceParams,
     );
+
+    if (payload.role == "viewer") {
```
Comment (Astitva877) [inline src/modules/workspace/services/workspace.service.ts:785]: take value from enum

File: src/modules/views/partials/footer.handlebars [hunk 4]
```diff
@@ -0,0 +1,57 @@
+{{!-- <div class="footer">
+    <table>
```
Comment (Astitva877) [inline src/modules/views/partials/footer.handlebars:2]: if not require can we remove this commented code.

**Review by Astitva877 — CHANGES_REQUESTED** [review]
Instead of making a separate for every mail we can define a common template function in @common module and can pass mail options for each mail.

File: src/modules/workspace/services/workspace.service.ts (hunk 3 above)
Comment (Abhishek9503) [reply src/modules/workspace/services/workspace.service.ts:785]: Resolved

File: src/modules/views/partials/footer.handlebars (hunk 4 above)
Comment (Abhishek9503) [reply src/modules/views/partials/footer.handlebars]: Resovled

**Comment by Abhishek9503** [comment]
> Instead of making a separate function for every mail we can define a common template function in @common module and can pass mail options for each mail.


Resovled

**Review by Astitva877 — APPROVED** [review]

**Review by itsmdasifraza — APPROVED** [review]

**Pull request merged**